formatter.go

```diff
@@ -472,6 +497,68 @@ func (f *DefaultFormatter) formatFloatValue(value float64, bits int) string {
 	}
 }
 
+func (f *DefaultFormatter) addDigitGrouping(numStr string) string {
```

Comment:
I suggest to rename this func to reformatNumber, and groupDigitsInString to insertDigitSeparator.

The first name suggests that we should call it when we're formatting any number, float or int. The caller doesn't bother what exactly it will do.

The second name correlates with the name of the DigitSeparator option.